build-logic/java-api-extractor/src/test/groovy/org/gradle/internal/tools/api/ApiClassExtractorTestSupport.groovy

```diff
@@ -116,25 +117,33 @@ class ApiClassExtractorTestSupport extends Specification {
     // you also update `ApiClassExtractorTest#target binary compatibility is maintained` with new assumptions.
     private static final String DEFAULT_TARGET_VERSION = '8'
 
-    protected ApiContainer toApi(Map<String, String> sources) {
+    protected ClassContainer toApi(Map<String, String> sources) {
         toApi(DEFAULT_TARGET_VERSION, [], sources)
     }
 
-    protected ApiContainer toApi(String targetVersion, Map<String, String> sources) {
+    protected ClassContainer toApi(String targetVersion, Map<String, String> sources) {
         toApi(targetVersion, [], sources)
     }
 
-    protected ApiContainer toApi(List<String> packages, Map<String, String> sources) {
+    protected ClassContainer toApi(List<String> packages, Map<String, String> sources) {
         toApi(DEFAULT_TARGET_VERSION, packages, sources)
     }
 
-    protected ApiContainer toApi(String targetVersion, List<String> packages, Map<String, String> sources) {
-        def dir = new File(temporaryFolder, 'out')
+    protected ClassContainer toApi(String targetVersion, List<String> packages, Map<String, String> sources) {
+        return compileTo(new File(temporaryFolder, 'api'), targetVersion, packages, sources, [])
+    }
+
+    protected ClassContainer compileTo(File dir, Map<String, String> sources, List<File> classpath) {
+        return compileTo(dir, DEFAULT_TARGET_VERSION, [], sources, classpath)
+    }
+
+    protected ClassContainer compileTo(File dir, String targetVersion, List<String> packages, Map<String, String> sources, List<File> classpath) {
         dir.mkdirs()
         def fileManager = compiler.getStandardFileManager(null, null, null)
+        fileManager.setLocation(StandardLocation.CLASS_PATH, classpath);
         def diagnostics = new DiagnosticCollector<JavaFileObject>()
         def task = compiler.getTask(
-            new OutputStreamWriter(new ByteArrayOutputStream()),
+            null,
```

Comment:
💭 This allows showing compiler output on `stderr`. Can be useful while debugging.